news/warnings.rst

```diff
@@ -0,0 +1,29 @@
+New features
+------------
+
+* Suppress numpy 1.20 deprecation warnings concerning the use of None as shape, np.float, np.int and np.bool
+
+Deprecations
+------------
+
+* Replace np.float by np.float64, np.int by int and np.bool by bool
```

Comment:
```suggestion
* <news item>
```

This section is for deprecating things in h5py's API, so no need to mention this.